### DISCUSSION TITLE
JPEG XL support

### DISCUSSION THREAD

**Discussion by nicopunktse** [Xanashi/Icaros#126] [Feature Requests]
It would be great to have support for JPEG XL (.jxl) files in Icaros.

Currently, this is the only format that noticeably lacks support when I browse my picture folder, aside from GIMP .xcf and PhotoScape X .psxproj files, which I guess are unlikely to receive support in the near future.

**Comment by Klaus1189** [comment]
Upload sample files on google drive or similar so Xanashi can take a look

**Reply by nicopunktse** [reply]
I have uploaded a JPEG XL (.jxl), a GIMP (.xcf), and a PhotoScape X (.psxprj) file to Google Drive at the following link:

[Test Image Files on Google Drive](https://drive.google.com/drive/folders/1mDDaSvjaD3jzn7GCr6xlQow3HXgsFd6x?usp=sharing)

Certainly, JPEG XL is paramount, and support for GIMP XCF would be beneficial. However, expectations for .psxprj support are low due to its niche status and proprietary nature.

**Comment by Xanashi** [comment]
Thank you for the test files! 

The good news is that JXL is already on the horizon. It will be added to Icaros at some point the near future.
I have it working internally, but there are a few kinks that need to be ironed out before I can publish it.

In the meantime, you can take a look [here](https://github.com/Xanashi/Icaros/issues/24) for alternatives.

I have looked into XCF, and it's unfortunately not a simple format to add to Icaros. So don't hold you breath on this one.
I can take a look at psxprj, and see what it would take to add that one.

**Reply by nicopunktse** [reply]
Thank you for your reply!

It's great to know that JXL support is on the horizon! I appreciate the interim solutions you've provided—I'll definitely check them out.

I understand that XCF is a complex format, so I'm not getting my hopes up.

Likewise for PSXPRJ really, considering it's a niche file format exclusive to PhotoScape X. Appreciate you looking into it though!

**Reply by nicopunktse** [reply]
I took a closer look at some .PSXPRJ files, it appears they are essentially zip-containers that store various layers as .png files, along with a flattened thumbnail.png. While the format itself isn't complex, I don't know if integrating support for it in Icaros would be straightforward though?

**Reply by Xanashi** [reply]
Since PSXPRJ is just a zip file with images, it is actually already supported in the current version of Icaros, you just have to drag one
of the PSXPRJ files into the thumbnailing page of IcarosConfig to enable it (or add the .psxprj filetype manually to the list).

With that said it is currently not recognized as an image format, but I'll make sure that it is in the upcoming release of Icaros.

**Comment by kamikazilucas** [comment]
i would also like jpeg xl since jpeg is nowhere near as good for storing high quality photos

**Comment by WolfExplode** [comment]
any progress on .jxl files?

**Reply by Xanashi** [reply]
Next thing I will look into finishing. (95% there)
